cluster/plan.go

```diff
@@ -0,0 +1,402 @@
+			"--root-ca-file=" + pki.GetCertPath(pki.CACertName),
+		},
+	}
+	if c.Authorization.Mode == services.RBACAuthorizationMode {
+		imageCfg.Cmd = append(imageCfg.Cmd, "--use-service-account-credentials=true")
+	}
+	hostCfg := &container.HostConfig{
+		VolumesFrom: []string{
+			services.SidekickContainerName,
+		},
+		Binds: []string{
+			"/etc/kubernetes:/etc/kubernetes",
+		},
+		NetworkMode:   "host",
+		RestartPolicy: container.RestartPolicy{Name: "always"},
+	}
+	for arg, value := range c.Services.KubeController.ExtraArgs {
+		cmd := fmt.Sprintf("--%s=%s", arg, value)
+		imageCfg.Entrypoint = append(imageCfg.Entrypoint, cmd)
+	}
+	healthCheck := v3.HealthCheck{
+		URL: fmt.Sprintf("%s%s:%d%s", services.HTTPProtoPrefix, services.HealthzAddress, services.KubeControllerPort, services.HealthzEndpoint),
+	}
+	return v3.Process{
+		ImageCfg:    imageCfg,
+		HostCfg:     hostCfg,
+		HealthCheck: healthCheck,
+	}
+}
+
+func (c *Cluster) BuildKubeletProcess(host *hosts.Host) v3.Process {
+	imageCfg := &container.Config{
+		Image: c.Services.Kubelet.Image,
+		Entrypoint: []string{"/opt/rke/entrypoint.sh",
+			"kubelet",
+			"--v=2",
+			"--address=0.0.0.0",
+			"--cluster-domain=" + c.ClusterDomain,
+			"--pod-infra-container-image=" + c.Services.Kubelet.InfraContainerImage,
+			"--cgroups-per-qos=True",
+			"--enforce-node-allocatable=",
+			"--hostname-override=" + host.HostnameOverride,
+			"--cluster-dns=" + c.ClusterDNSServer,
+			"--network-plugin=cni",
+			"--cni-conf-dir=/etc/cni/net.d",
+			"--cni-bin-dir=/opt/cni/bin",
+			"--resolv-conf=/etc/resolv.conf",
+			"--allow-privileged=true",
+			"--cloud-provider=",
+			"--kubeconfig=" + pki.GetConfigPath(pki.KubeNodeCertName),
+			"--require-kubeconfig=True",
+			"--fail-swap-on=" + strconv.FormatBool(c.Services.Kubelet.FailSwapOn),
+		},
+	}
+	hostCfg := &container.HostConfig{
+		VolumesFrom: []string{
+			services.SidekickContainerName,
+		},
+		Binds: []string{
+			"/etc/kubernetes:/etc/kubernetes",
+			"/usr/libexec/kubernetes/kubelet-plugins:/usr/libexec/kubernetes/kubelet-plugins",
+			"/etc/cni:/etc/cni:ro",
+			"/opt/cni:/opt/cni:ro",
+			"/etc/resolv.conf:/etc/resolv.conf",
+			"/sys:/sys",
+			"/var/lib/docker:/var/lib/docker:rw",
+			"/var/lib/kubelet:/var/lib/kubelet:shared",
+			"/var/run:/var/run:rw",
+			"/run:/run",
+			"/etc/ceph:/etc/ceph",
+			"/dev:/host/dev",
+			"/var/log/containers:/var/log/containers",
+			"/var/log/pods:/var/log/pods"},
+		NetworkMode:   "host",
+		PidMode:       "host",
+		Privileged:    true,
+		RestartPolicy: container.RestartPolicy{Name: "always"},
+	}
+	for arg, value := range c.Services.Kubelet.ExtraArgs {
+		cmd := fmt.Sprintf("--%s=%s", arg, value)
+		imageCfg.Entrypoint = append(imageCfg.Entrypoint, cmd)
+	}
+	healthCheck := v3.HealthCheck{
+		URL: fmt.Sprintf("%s%s:%d%s", services.HTTPSProtoPrefix, services.HealthzAddress, services.KubeletPort, services.HealthzEndpoint),
```

Comment:
can you create one unified function to return the healthcheck URL and send parameters to it